### PR TITLE
Integrate identity-spring-boot-starter in Zeebe

### LINKED ISSUE #15389: Configure `identity-sdk` through `identity-spring-boot-starter` in the Gateway
**Is your feature request related to a problem? Please describe.**
Camunda Identity now provides an `identity-spring-boot-starter` which, in turn, provides an `IdentityConfiguration` Spring bean.

The goal is to decouple the Identity configuration properties from the Zeebe configuration classes. The `identity-spring-boot-starter` should create an `IdentityConfiguration` Spring bean, which in turn can be used to configure the `identity-sdk` inside Zeebe. 

The outcome will be that Zeebe doesn't need to add new Identity configuration properties every time Identity adds them.

**Describe the solution you'd like**

- [x] The `IdentityInterceptor` class should create an `Identity` instance using an `IdentityConfiguration` instance ([src](https://github.com/camunda/zeebe/blob/303b0d7bbc60440fee8baaf943ab020d49783d38/gateway/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/IdentityInterceptor.java#L37-L39)).
- [x] The `Gateway` class should pass an `IdentityConfiguration` instance to the `IdentityInterceptor` constructor ([src](https://github.com/camunda/zeebe/blob/b3c3397f10b8e96d12af053ff72d616f80264dca/gateway/src/main/java/io/camunda/zeebe/gateway/Gateway.java#L323-L328)).
- [x] The `StandaloneGateway` class should `Autowire` an `IdentityConfiguration` bean and pass it along to the `Gateway` constructor ([src](https://github.com/camunda/zeebe/blob/44769129dbfbdee7b44889780215d20e395453f2/dist/src/main/java/io/camunda/zeebe/gateway/StandaloneGateway.java#L101-L103)).
- [x] The `EmbeddedGatewayService` class should pass an `IdentityConfiguration` instance to the `Gateway` constructor ([src](https://github.com/camunda/zeebe/blob/de669e90aa74a0330f6e17f07de0b14aa430a2cc/broker/src/main/java/io/camunda/zeebe/broker/system/EmbeddedGatewayService.java#L36-L37)).
   - The `IdentityConfiguration` bean should be passed through the `StandaloneBroker` class ([src](https://github.com/camunda/zeebe/blob/4c4e7d2404f00045e5ac3d033e007e30b063b14c/dist/src/main/java/io/camunda/zeebe/broker/StandaloneBroker.java#L82)).
- [x] The `MultiTenancyOverIdentityIT` test should configure Identity using an `IdentityConfiguration` bean ([src](https://github.com/camunda/zeebe/blob/d98d04cbe6ab675c981aea808566264f8552d647/qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java#L210-L213)).

**Describe alternatives you've considered**
- None. Zeebe needs to integrate the `identity-spring-boot-starter` in order to decouple the Identity configuration properties from Zeebe.

**Additional context**
* https://github.com/camunda/product-hub/issues/1849
* https://github.com/camunda-cloud/identity/issues/2162

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Decouple the `identity-sdk` configuration properties from Zeebe by integrating the `identity-spring-boot-autoconfiguration` dependency, which will provide a `IdentityConfiguration` bean if the `Identity` class is
available on the classpath.

## Related issues

closes #15389 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: broker/pom.xml [hunk 1]
```diff
@@ -146,6 +146,11 @@
       <artifactId>netty-common</artifactId>
     </dependency>
 
+    <dependency>
```
Comment (npepinpe) [inline broker/pom.xml:149]: :question: This is added only so we can pass it down to the gateway config, right?

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationIdentityIT.java [hunk 2]
```diff
@@ -124,9 +125,9 @@ static void beforeAll() {
             cfg -> {
               final var auth = cfg.getGateway().getSecurity().getAuthentication();
               auth.setMode(AuthMode.IDENTITY);
-              auth.getIdentity().setIssuerBackendUrl(getKeycloakRealmAddress());
-              auth.getIdentity().setAudience(ZEEBE_CLIENT_AUDIENCE);
             })
+        .withProperty("camunda.identity.issuerBackendUrl", getKeycloakRealmAddress())
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationIdentityIT.java:129]: :rocket:

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java [hunk 3]
```diff
@@ -75,10 +75,9 @@
 @AutoCloseResources
 public class MultiTenancyOverIdentityIT {
 
+  public static final String IDENTITY_SNAPSHOT_TAG = "SNAPSHOT";
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java:78]:
```suggestion
  private static final String IDENTITY_SNAPSHOT_TAG = "SNAPSHOT";
```

**Review by npepinpe — APPROVED** [review]
:+1: 

I'm not super happy about having to pass everything through the broker module just to end up in the gateway, but I don't have a better solution that doesn't involve refactoring a lot :smile:

File: dist/src/main/config/broker.standalone.yaml.template [hunk 4]
```diff
@@ -82,6 +82,8 @@
           # This setting can also be overridden using the environment variable ZEEBE_BROKER_GATEWAY_SECURITY_AUTHENTICATION_MODE.
           # mode: none
 
+          # WARNING: These settings are deprecated! Please use the corresponding Camunda Identity
+          # properties or the environment variables defined here: https://docs.camunda.io/docs/self-managed/identity/deployment/configuration-variables/
```
Comment (korthout) [inline dist/src/main/config/broker.standalone.yaml.template:86]: 💭 Now that we're deprecating these we must also adjust the documentation https://docs.camunda.io/docs/next/self-managed/zeebe-deployment/configuration/gateway-config/#zeebegatewaymultitenancy

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java [hunk 5]
```diff
@@ -206,28 +201,16 @@ static void init() throws Exception {
...
             })
+        .withProperty(
+            "camunda.identity.baseUrl",
+            "http://%s:%d".formatted(IDENTITY.getHost(), IDENTITY.getMappedPort(8080)))
+        .withProperty(
+            "camunda.identity.issuerBackendUrl",
+            "http://%s:%d%s"
+                .formatted(
+                    KEYCLOAK.getHost(), KEYCLOAK.getMappedPort(8080), KEYCLOAK_PATH_CAMUNDA_REALM))
+        .withProperty("camunda.identity.audience", ZEEBE_CLIENT_AUDIENCE)
```
Comment (korthout) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java:213]: 👍 Nice

**Review by korthout — APPROVED** [review]
I did not have time for a full detailed review, but on a quick read through it looks good to me 👍 

If you need a more thorough review, please request it from someone else if possible

File: dist/src/main/config/broker.standalone.yaml.template (hunk 4 above)
Comment (koevskinikola) [reply dist/src/main/config/broker.standalone.yaml.template:86]: I already created a PR for adjusting the docs [here](https://github.com/camunda/camunda-docs/pull/2981), but good point, I need to expand it to the MT section as well.

File: gateway/src/main/java/io/camunda/zeebe/gateway/Gateway.java [hunk 6]
```diff
@@ -321,15 +334,21 @@ private ServerServiceDefinition applyInterceptors(final BindableService service)
     interceptors.add(new ContextInjectingInterceptor(queryApi));
     interceptors.add(MONITORING_SERVER_INTERCEPTOR);
     if (AuthMode.IDENTITY == gatewayCfg.getSecurity().getAuthentication().getMode()) {
-      interceptors.add(
-          new IdentityInterceptor(
-              gatewayCfg.getSecurity().getAuthentication().getIdentity(),
-              gatewayCfg.getMultiTenancy()));
+      final var zeebeIdentityCfg = gatewayCfg.getSecurity().getAuthentication().getIdentity();
+      if (isZeebeIdentityConfigurationNotNull(zeebeIdentityCfg)) {
+        interceptors.add(new IdentityInterceptor(zeebeIdentityCfg, gatewayCfg.getMultiTenancy()));
```
[the remark this answers is not in the document]
Comment (koevskinikola) [reply gateway/src/main/java/io/camunda/zeebe/gateway/Gateway.java:339]: @npepinpe I'm already doing that [here](https://github.com/camunda/zeebe/pull/15434/files/bcfc5997340a6251b09ed6581d0c9c94b0a94101#diff-6783f1e71cace45bffaa5446fc6bde1f63deb6ae0ff99fd28eeae7e659881253R85), and I opened a [PR to deprecate the docs](https://github.com/camunda/camunda-docs/issues/2980). 

Do you think something else is needed?

[the remark this answers is not in the document]
Comment (npepinpe) [reply gateway/src/main/java/io/camunda/zeebe/gateway/Gateway.java:339]: Yes, a log warning users about it, and an update note about it in the docs. How many of us actually read the docs/config template files? ;)

If I upgrade the app, I don't typically re-read the docs or config.

**Pull request merged**